.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,71 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+name: "CodeQL"
+
+on:
+  push:
+    branches: [master, proposed/3.0.0]
```

Comment:
Is there value in checking the v3 branch if it's likely not going to be released?

Reply:
Better question: Is there value in keeping it around if it's not going to be released?

Reply:
Removed it in the second commit as well

Reply:
I think there's a lot of valuable work on the proposed/3.0.0 branch that *could* be released at some point. I was keeping it up to date until we had some of the crazier commit history changes in 2018. I think we can get it back in sync but it's going to take a pretty tedious merge. Alternatively, we could hand pick the relevant feature commits, but that's also a big time investment.

That said, I don't think there's any value in checking the current proposal branch here.

Reply:
Frankly, I think there's a lot of internal refactoring that could make a 3.0 easier to release and maintain long-term. It's something I've thought about for over a year